src/event.rs

```diff
@@ -549,42 +549,7 @@ where
 						}
 						payment_preimage
 					},
-					PaymentPurpose::Bolt12RefundPayment {
-						payment_preimage,
-						payment_secret,
-						..
-					} => {
-						let payment = PaymentDetails {
-							id: payment_id,
-							kind: PaymentKind::Bolt12Refund {
-								hash: Some(payment_hash),
-								preimage: payment_preimage,
-								secret: Some(payment_secret),
-							},
-							amount_msat: Some(amount_msat),
-							direction: PaymentDirection::Inbound,
-							status: PaymentStatus::Pending,
-						};
-						match self.payment_store.insert(payment) {
-							Ok(false) => (),
-							Ok(true) => {
-								log_error!(
-									self.logger,
-									"Bolt12RefundPayment with ID {} was previously known",
-									payment_id,
-								);
-								debug_assert!(false);
-							},
-							Err(e) => {
-								log_error!(
-									self.logger,
-									"Failed to insert payment with ID {}: {}",
-									payment_id,
-									e
-								);
-								debug_assert!(false);
-							},
-						}
```

Comment:
If possible, could you remove this from the earlier commit since it was not needed.